crates/sc-consensus-subspace/src/archiver.rs

```diff
@@ -687,6 +688,16 @@ where
             for archived_segment in archiver.add_block(encoded_block, block_object_mappings) {
                 let segment_header = archived_segment.segment_header;
 
+                if let Err(error) =
```

Comment:
Unrelated to this change(github doesn't allow comments on unchanged lines) ..

L612 above (`block_number.checked_sub(&confirmation_depth_k.into()` ...). This looks like an edge case where we won't archive blocks when the u32 wraps around. When this happens, there could be a slow path that just walks back the parent hashes to get to the block k_conf_depth old). 

Or we could opportunistically cache the block hashes as import notifications come in (this won't work on restart or reorg), then fall back to the slow path

Reply:
`u32` will never wrap around, our block number type is `u32`

Reply:
Is it because mainnet needs to be running for a very loong time (at 1 block per 6 sec) to get there? nice problem to have though :-

Reply:
Exactly. Even if blocks are produced every second running out of `u32` will not be an issue to worry for quite a few decades. I hope Subspace lives long enough for this to be a real issue.